lib/DateTime/Format/Strptime.pm

```diff
@@ -607,8 +607,19 @@ sub _token_re_for {
         }
 
         if ( $args->{time_zone_offset} ) {
+            my ( $sign, $hours, $minutes ) = ( '+', 0, 0 );
+
+            if ( $args->{time_zone_offset} ne 'Z' ) {
+                $args->{time_zone_offset} =~ s/://;
+                ( $sign, $hours, $minutes ) = unpack 'A(A2)*',
```

Comment:
I feel like `unpack` instead of a regex is a little more confusing than needed. I prefer to only use `unpack` when dealing with binary data.